admin_pages/venues/espresso_events_Venues_Hooks.class.php

```diff
@@ -131,21 +126,73 @@ public function venue_metabox()
 
         $template_args['venues'] = $venues;
         $template_args['evt_venue_id'] = $evt_venue_id;
-        $template_args['venue_selector'] = EEH_Form_Fields::select_input(
-            'venue_id',
+        $venue_selector = new EE_Select_Input(
             $ven_select,
-            $evt_venue_id,
-            'id="venue_id"'
+            array(
+                'html_name'  => 'venue_id',
+                'html_id'    => 'venue_id',
+                'html_class' => 'wide',
+                'default'    => $evt_venue_id ? $evt_venue_id : '0'
+            )
         );
-        $template_args['enable_for_gmap'] = EEH_Form_Fields::select_input(
-            'enable_for_gmap',
-            $values,
-            is_object($evt_venue)
-                ? $evt_venue->enable_for_gmap() : null,
-            'id="enable_for_gmap"'
+        $template_args['venue_selector'] = $venue_selector->get_html_for_input();
+        $enable_for_gmap = new EE_Yes_No_Input(
+            array(
+                'html_name'  => 'enable_for_gmap',
+                'html_id'    => 'enable_for_gmap',
+                'default'    => is_object($evt_venue) ? $evt_venue->enable_for_gmap() : false
```

Comment:
This would be a good opportunity to make this ternary a bit more robust by checking if `$evt_venue instanceof EE_Venue` instead of `is_object`.

Reply:
+1